Dockerfile

```diff
@@ -0,0 +1,85 @@
+FROM debian:10
```

Comment:
where and what are you using this for?  Don't we have it already inside https://github.com/Azure/azure-sdk-for-cpp/tree/main/.devcontainer ??

Reply:
the thing here is that in order to build a docker image it has to be in the root folder of the "build context" aka the root folder where you are targeting. There are some small changes from the dev docker -- no vcpkg -- 
This image can then be used as base for the Docker.src image. I split it in two files in order to speed up testing build docker image

Reply:
>  it has to be in the root folder

I am not super sure about this.

But, I would like to understand better what we want to do with this file and docker.  Can you elaborate about it?  Or maybe create a doc for users/customers for how they can use this?

I am a bit confused if this is only used to test what you are enabling, or if this is actually required/part of the feature.